plugins/catalog/src/components/CatalogTable/CatalogTable.tsx

```diff
@@ -249,6 +253,7 @@ export const CatalogTable = (props: CatalogTableProps) => {
       }}
       components={{
         Toolbar: CatalogTableToolbar,
+        ...(components ?? {}),
```

Comment:
I think `CatalogTableToolbar` shouldn't be used here as it isn't supposed to be used for client side paginated table.
Would removing the line above fix your issue @kuangp?

Reply:
Yes! that would fix it for us since we don't currently use server side pagination - I'll remove this as a "workaround" for now however the original issue I described will still exist for those who do use server side pagination and wish to use built-in Material Table toolbar features (in addition to the inconsistency between the table impls since it would've been nice to have search query URL persistence here as well) but I guess than can be revisited in a subsequent change